src/Hl7.Fhir.Base/Model/CodeOfT.cs

```diff
@@ -104,7 +104,7 @@ public override IEnumerable<ValidationResult> Validate(ValidationContext validat
                 return baseResults;
             else
             {
-                var result = COVE.INVALID_CODED_VALUE.AsResult(validationContext, ObjectValue, EnumUtility.GetName<T>());
+                var result = COVE.INVALID_CODED_VALUE(validationContext, ObjectValue, EnumUtility.GetName<T>()).AsResult();
```

Comment:
This would have to become  

`INVALID_CODED_VALUE(ObjectValue,EnumUtility.GetName<T>).AsResult(validationContext)`, as it is important for me to keep the domain of the CodedExceptions separate from the ValidationResult class.  In other words: I want to be able to work with CodedValidationExceptions without needing to know about a `ValidationContext`.  E.g. if we ever move away from the .NET attribute validation, I'd still like to be able to use the same CodedValidationException class. We cannot do that if the constructor is using it.

Reply:
In fact, it would make sense to move the AsResult() totally away from the CodedValidationException by turning this into an extension method on CodedValidationException.

Reply:
This was resolved putting the validation context in both functions and not storing it between calls.